toolkit/docs/security/read-only-roots.md

```diff
@@ -0,0 +1,111 @@
+Read-Only Roots
+===
+
+- [dm-verity Read Only Roots](#dm-verity-read-only-roots)
+  - [Creating Verity Disks](#Creating-Verity-Disks)
+  - [Writable Areas](#Writable-Areas)
+  - [Root of Trust](#Root-of-Trust)
+  - [Users, Machine ID, Misc. Configuration](#Users-Machine-ID-Misc-Configuration)
+  - [Forward Error Correction (FEC)](#Forward-Error-Correction-(FEC))
+  - [Overhead](#Overhead)
+  - [Debugging](#Debugging)
+  - [ISO Installers](#ISO-Installers)
+
+## dm-verity Read Only Roots
+`dm-verity` is a Device Mapper target which creates read-only partitions with built-in integrity checking. This means a verity partition can detect (and if FEC is enabled, fix) changes to the data in the partition.
+
+A Merkle hash tree is created from the measurements of each block in the partition (Hash each 4k block, then hash those hashes. Repeat until you have a single hash). The root node of the tree, a single hash, is called the root hash.
+
+When data is read from disk the blocks are hashed and checked against the hash tree to detect any changes. Any change to a block will modify its branch of the tree, all the way down to the root hash. By securing the root hash the entire partition can be secured.
+
+### Creating Verity Disks
+Read-only roots can be enabled using the `"ReadOnlyVerityRoot"` key in the image configuration JSON file (See [imageconfig.md](../formats/imageconfig.md#ReadOnlyVerityRoot) for details)
+Since the read-only partition becomes unmodifiable as soon as its measured, the snapshot must be taken at the very end of the build process. The `imager` tool switches the root to read-only mode, then measures it into a hash tree, FEC data, and root hash using the `veritysetup` tool. These files are then placed into the `initramfs`.
```

Comment:
```suggestion
Since the read-only partition becomes unmodifiable as soon as it's measured, the snapshot must be taken at the very end of the build process. The `imager` tool switches the root to read-only mode, then uses the `veritysetup` tool to measure the root into a hash tree, FEC data, and compute the root hash. These files are then placed into the `initramfs`.
```
nit - clarity